### PR TITLE
[agent docs]add reporting guide

### PR DESCRIPTION
**Pull request by yiyixuxu**

### CODE REVIEW

File: .ai/reporting.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
+# Reporting bugs and performance claims
+
+For issues and PR descriptions on this repo, from humans and agents alike.
+
+## Bug reports
+
+A reproduction is **what you were doing when it broke, with everything unnecessary removed** — start from the real failing situation and delete, don't build a clean synthetic case from scratch. The test: can someone paste it into a terminal and see the same failure, without first accepting your theory of the cause? A script that demonstrates your theory is not a reproduction — it can pass while the real bug is still there.
```
Comment (sayakpaul) [inline .ai/reporting.md]: We won't probably be pasting the code into the terminal but rather create a Python file out of the code snippet and run it?

File: .ai/reporting.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+- Keep the real model, settings, and dtype. A repro that downloads weights and takes two minutes is worth more than a fast synthetic one.
```
Comment (sayakpaul) [inline .ai/reporting.md]: But we should also ask it to strive to be minimal even in that regard. For example, just using the bare minimum dependencies, and no externalities at all (like invoking diffusers from a servlet for example). Thoughts?

**Review by sayakpaul — APPROVED** [review]
Super nice!

File: .ai/reporting.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
+# Reporting bugs and performance claims
+
+For issues and PR descriptions on this repo, from humans and agents alike.
```
Comment (stevhliu) [inline .ai/reporting.md]:
```suggestion
Use this guide when writing bug reports or performance claims in issues and pull requests.
```

File: .ai/reporting.md (hunk 1 above)
Comment (stevhliu) [inline .ai/reporting.md]:
```suggestion
A reproduction is the smallest version of the real workflow that still fails. Start with what you were doing when the failure occurred, then remove anything unrelated. Someone should be able to run it and see the same failure without first accepting your theory about its cause. A script that tests your theory is not a reproduction. It may pass even when the real workflow still fails.
```

File: .ai/reporting.md (hunk 2 above)
Comment (stevhliu) [inline .ai/reporting.md]:
```suggestion
- Keep the real model, settings, and dtype. A slower repro is better than a faster one that changes the behavior.
```

File: .ai/reporting.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+```python
+# "enable_gqa + attn_mask falls back to the math backend and materializes an S x S score matrix"
+q = torch.randn(1, 24, 8100, 128, device="cuda", dtype=torch.bfloat16)
+k = v = torch.randn(1, 6, 8100, 128, device="cuda", dtype=torch.bfloat16)
+mask = torch.ones(1, 1, 8100, 8100, dtype=torch.bool, device="cuda")
+F.scaled_dot_product_attention(q, k, v, attn_mask=mask, enable_gqa=True)
+print(torch.cuda.max_memory_allocated())
+```
+
+The difference: a maintainer understands the first script at a glance — it's an ordinary pipeline call — while the second is not, and it only demonstrates the reporter's theory: it can go green while the real pipeline still dies on a 24GB card.
```
Comment (stevhliu) [inline .ai/reporting.md]:
```suggestion
The first script exposes the real pipeline call. The second only tests the proposed mechanism, so it may pass even when the pipeline still fails. Keep the hypothesis separate from the reproduction.
```

File: .ai/reporting.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+print(torch.cuda.max_memory_allocated())
+```
+
+The difference: a maintainer understands the first script at a glance — it's an ordinary pipeline call — while the second is not, and it only demonstrates the reporter's theory: it can go green while the real pipeline still dies on a 24GB card. 
+
+For structure, follow the [bug report template](../.github/ISSUE_TEMPLATE/bug-report.yml).
+
+## Performance claims
+
+Report **the end-to-end number only**: wall clock for the full pipeline call, before vs. after, on the same hardware, dtype, and seed. That's the number we decide with — if we want op-level detail, we'll ask.
```
Comment (stevhliu) [inline .ai/reporting.md]:
```suggestion
Lead with the end-to-end result: measure the full pipeline call before and after the change on the same hardware, dtype, and seed. Only include lower-level measurements when requested.
```

File: .ai/reporting.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+Attach the script you measured with, please keep it as simple as possible (see `benchmarks/benchmarking_utils.py`):
```
Comment (stevhliu) [inline .ai/reporting.md]:
```suggestion
Attach the measurement script and keep it as simple as possible. See `benchmarks/benchmarking_utils.py` for the repository's benchmarking helper.
```

File: .ai/reporting.md [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+def benchmark_fn(f, *args, **kwargs):
+    t0 = benchmark.Timer(stmt="f(*args, **kwargs)", globals={"args": args, "kwargs": kwargs, "f": f}, num_threads=1)
+    return f"{t0.blocked_autorange().mean:.3f}s"
+
+pipe(**call_kwargs)  # warmup
+print(benchmark_fn(pipe, **call_kwargs))
+print(f"peak memory: {torch.cuda.max_memory_allocated() / 1024**3:.2f}GB")
+```
+
+State the setup (GPU, dtype, torch/diffusers versions, attention backend) and the exact call (model id, resolution/duration, steps, batch size). One-shot timings are noise.
```
Comment (stevhliu) [inline .ai/reporting.md]:
```suggestion
State the setup (GPU, dtype, torch/diffusers versions, attention backend) and the exact call (model id, resolution/duration, steps, batch size). One-shot timings are noisy.
```

**Review by stevhliu — APPROVED** [review]
thanks!
